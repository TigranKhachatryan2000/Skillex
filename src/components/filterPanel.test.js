import { render, screen, fireEvent } from "@testing-library/react";

import FilterPanel from "./filterPanel";

describe("FilterPanel", () => {
  let setFiltersMock;
  let handlePriceRangeChangeMock;

  beforeEach(() => {
    setFiltersMock = jest.fn();
    handlePriceRangeChangeMock = jest.fn();
  });

  test("renders filter options correctly", () => {
    const { asFragment } = render(
      <FilterPanel
        filters={{ rating: 0 }}
        setFilters={setFiltersMock}
        handlePriceRangeChange={handlePriceRangeChangeMock}
      />
    );

    expect(asFragment()).toMatchSnapshot();

    expect(screen.getByRole("slider")).toBeInTheDocument();
    expect(screen.getByRole("spinbutton")).toBeInTheDocument();
    expect(screen.getByText("Filter by Brand")).toBeInTheDocument();
    expect(screen.getByText("Filter by Price")).toBeInTheDocument();
    expect(screen.getByText("Filter by Rating")).toBeInTheDocument();
    expect(screen.getByText("Filter by Category")).toBeInTheDocument();
  });

  test("calls handlePriceRangeChange when price is adjusted", () => {
    const { asFragment } = render(
      <FilterPanel
        filters={{ rating: 0 }}
        setFilters={setFiltersMock}
        handlePriceRangeChange={handlePriceRangeChangeMock}
      />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.change(screen.getByRole("slider"), { target: { value: 250 } });
    expect(handlePriceRangeChangeMock).toHaveBeenCalledWith(0, 250);
  });

  test("calls setFilters when rating is changed", () => {
    const { asFragment } = render(
      <FilterPanel
        filters={{ rating: 0 }}
        setFilters={setFiltersMock}
        handlePriceRangeChange={handlePriceRangeChangeMock}
      />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.change(screen.getByRole("spinbutton"), {
      target: { value: 4.5 },
    });
    expect(setFiltersMock).toHaveBeenCalledWith(expect.any(Function));

    const updatedFilters = setFiltersMock.mock.calls[0][0]({});
    expect(updatedFilters.rating).toBe("0");
  });
});

afterAll(() => jest.clearAllMocks());
