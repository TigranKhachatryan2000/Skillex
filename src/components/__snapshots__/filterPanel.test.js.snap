// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterPanel calls handlePriceRangeChange when price is adjusted 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Filter by Category
    </h3>
    <label
      for="category"
    >
      Category
    </label>
    <select
      id="category"
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="Electronics"
      >
        Electronics
      </option>
      <option
        value="Footwear"
      >
        Footwear
      </option>
      <option
        value="Clothing"
      >
        Clothing
      </option>
    </select>
    <h3>
      Filter by Brand
    </h3>
    <label
      for="brand"
    >
      Brand
    </label>
    <select
      id="brand"
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="Brand A"
      >
        Brand A
      </option>
      <option
        value="Brand B"
      >
        Brand B
      </option>
    </select>
    <h3>
      Filter by Price
    </h3>
    <input
      max="500"
      min="0"
      step="10"
      type="range"
    />
    <h3>
      Filter by Rating
    </h3>
    <input
      max="5"
      min="0"
      step="0.1"
      type="number"
      value="0"
    />
  </div>
</DocumentFragment>
`;

exports[`FilterPanel calls setFilters when rating is changed 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Filter by Category
    </h3>
    <label
      for="category"
    >
      Category
    </label>
    <select
      id="category"
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="Electronics"
      >
        Electronics
      </option>
      <option
        value="Footwear"
      >
        Footwear
      </option>
      <option
        value="Clothing"
      >
        Clothing
      </option>
    </select>
    <h3>
      Filter by Brand
    </h3>
    <label
      for="brand"
    >
      Brand
    </label>
    <select
      id="brand"
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="Brand A"
      >
        Brand A
      </option>
      <option
        value="Brand B"
      >
        Brand B
      </option>
    </select>
    <h3>
      Filter by Price
    </h3>
    <input
      max="500"
      min="0"
      step="10"
      type="range"
    />
    <h3>
      Filter by Rating
    </h3>
    <input
      max="5"
      min="0"
      step="0.1"
      type="number"
      value="0"
    />
  </div>
</DocumentFragment>
`;

exports[`FilterPanel renders filter options correctly 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Filter by Category
    </h3>
    <label
      for="category"
    >
      Category
    </label>
    <select
      id="category"
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="Electronics"
      >
        Electronics
      </option>
      <option
        value="Footwear"
      >
        Footwear
      </option>
      <option
        value="Clothing"
      >
        Clothing
      </option>
    </select>
    <h3>
      Filter by Brand
    </h3>
    <label
      for="brand"
    >
      Brand
    </label>
    <select
      id="brand"
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="Brand A"
      >
        Brand A
      </option>
      <option
        value="Brand B"
      >
        Brand B
      </option>
    </select>
    <h3>
      Filter by Price
    </h3>
    <input
      max="500"
      min="0"
      step="10"
      type="range"
    />
    <h3>
      Filter by Rating
    </h3>
    <input
      max="5"
      min="0"
      step="0.1"
      type="number"
      value="0"
    />
  </div>
</DocumentFragment>
`;
